### ISSUE TITLE
Setup issues with Ubuntu 22 (within Windows Subsystem for Linux 2)

### ISSUE THREAD

**Issue by Arpanio** [CivicTechTO/toronto-bids#12]
Setup via `make` doesn't fully work with a fresh Ubuntu 22 VM (installed via Windows Subsystem for Linux 2).

I got a number of errors and was able to resolve them to be able to get `jupyter-notebook` running. None of the notebooks have been tested yet.

Error message (there were more than 1):
> arpanio@DESKTOP-CF2G2MB:~/projects/toronto-bids$ make run-jupyter-notebook
> python3.9 -m venv venv
> Error: Command '['/home/arpanio/projects/toronto-bids/venv/bin/python3.9', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
> make: *** [Makefile:23: venv] Error 1

Steps to resolve with the links that worked for me (may not be exhaustive as I didn't keep good notes while working on this):

1. [Setup Python 3.9](https://linuxize.com/post/how-to-install-python-3-9-on-ubuntu-20-04/). The current default version that ships with Ubuntu is 3.10.
2. [Setup pip for python 3.9](https://stackoverflow.com/a/71726397).
3. [Install Google Chrome](https://adamtheautomator.com/install-google-chrome-for-ubuntu/#Installing_Google_Chrome_for_Ubuntu_via_the_Terminal)

Can we incorporate all of this setup into `make`?

**Comment by sonalranjit** [comment]
Yeah we should definitely bump up the python version, i set `3.9` because that was the one that worked for my machine. But a another LTS version like 3.10 should be fine.

**Comment by Arpanio** [comment]
Instead of bumping up the python version, could we do the following?

- Detect if `python3.9` exists
- If it appears like it doesn't, give a prompt to users to update the path to v3.9 and inform that `python3.9` will be used as the alias for it
- If there is no v3.9, then download and install it setting `python3.9` as the alias
- Setup `pip` for `python3.9` (or ask for the path to it if the user indicated earlier that they already have python3.9 installed somewhere non-standard)
- Detect if Google Chrome exists in the default location. If not, then prompt the user for the path to Google Chrome.
- If the user indicates that there is no Chrome installation, get permission to install at the default location and install

I think the above steps will let someone setup python v3.9 on their Linux system without overwriting other stuff that they might need for other projects. Tomorrow if the LTS version changes to 3.11 then our `make` still works.

Another question is when do we want to work on this? At the moment, I think we've overcommitted to ensuring a reproducible setup and build, and under-committed to building features. We should keep this bug for documentation purposes and focus more on building the features.

**Comment by AcharyaS97** [comment]
I'm wondering if we could use this issue as a catch-all for bugs with running the scraper on WSL. The issues Arpan mentioned above, they're generic to the makefile regardless of how it runs no?

**Comment by alexwolson** [comment]
Closing: this is genuinely fixed, not just obsolete.

The reported failure was `make run-jupyter-notebook` → `python3.9 -m venv venv` → `ensurepip` failing, because Python 3.9 doesn't ship with Ubuntu 22. The suggested resolution steps were: install Python 3.9 manually, wire up pip for it, install Chrome.

All three causes are gone:

1. **The Python version pain is handled by the tooling.** The project moved to [uv](https://docs.astral.sh/uv/), which pins and installs the interpreter itself (3.12) — no system Python required, and no manual install steps on any distro. `uv sync` is the whole setup.
2. **The Makefile was deleted** in #60. Every target in it was broken, most for over two years — including `run-jupyter-notebook`, which was `.PHONY` with no target defined, so it silently did nothing and exited 0. That's why it "didn't fully work": there was nothing to run. `uv run tb` needs no wrapper.
3. **Chrome/Selenium is no longer needed.** The scraper is HTTP-only against public JSON/CSV/HTML endpoints. (One opt-in command, `tb enrich-council`, does drive a browser, but it's outside the default pipeline and not required for setup.)

To answer the original question — *"Can we incorporate all of this setup into make?"* — the answer turned out to be that the setup itself was the problem. It's now:

```shell
cd scrapers
uv sync
uv run pytest   # 142 passing, offline
```

Thanks for the detailed report — the version-pinning pain it documented is exactly what uv removes.
